core/service/src/lib.rs

```diff
@@ -82,6 +82,14 @@ pub struct Service<Components: components::Components> {
 	keystore: Keystore,
 	exit: ::exit_future::Exit,
 	signal: Option<Signal>,
+	/// Sender for futures that must be spawned as background tasks.
+	to_spawn_tx: mpsc::UnboundedSender<Box<dyn Future<Item = (), Error = ()> + Send>>,
+	/// Receiver for futures that must be spawned as background tasks.
+	to_spawn_rx: Mutex<mpsc::UnboundedReceiver<Box<dyn Future<Item = (), Error = ()> + Send>>>,
```

Comment:
Why do we need to put this into a mutex? (below the same question)

Reply:
*sighs* Because we have tests that use an `Arc<Service>`.

Reply:
Ahh.. Do you know which one are doing this?

(We should at least file an issue to solve this)

Reply:
Here: https://github.com/paritytech/substrate/blob/96e17dc57fc7e6744c6c6f824f9c35e31350f709/core/service/test/src/lib.rs#L47-L49

Reply:
This looks like it should be a `Box` and not an arc.